sensor/kubernetes/listener/resources/deployment_store.go

```diff
@@ -20,6 +20,21 @@ type DeploymentStore struct {
 	deployments map[string]*deploymentWrap
 }
 
+// ReconcileDelete is called after Sensor reconnects with Central and receives its state hashes.
+// Reconciliacion ensures that Sensor and Central have the same state by checking whether a given resource
+// shall be deleted from Central.
+func (ds *DeploymentStore) ReconcileDelete(resType, resID string, _ uint64) (string, error) {
+	if resType != "Deployment" {
+		return "", nil
+	}
+	depl := ds.Get(resID)
```

Comment:
Similar to the pod store comment, let's avoid making the `Get` call which would incur unnecessary clone. 

1) Lock the store
2) Use `getWrapNoLock`
3) If not nil, return the ID.